src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -320,10 +320,19 @@ bool SystemDictionaryShared::check_for_exclusion_impl(InstanceKlass* k) {
     }
   }
 
-  if (k->is_hidden() && !is_registered_lambda_proxy_class(k)) {
-    ResourceMark rm;
-    log_debug(cds)("Skipping %s: Hidden class", k->name()->as_C_string());
-    return true;
+  if (k->is_hidden()) {
+    if (!is_registered_lambda_proxy_class(k)) {
+      ResourceMark rm;
+      log_debug(cds)("Skipping %s: Hidden class", k->name()->as_C_string());
+      return true;
+    } else {
+      DumpTimeClassInfo* info = _dumptime_table->get(k);
+      assert(info != NULL, "sanity");
+      if (check_for_exclusion(info->nest_host(), NULL)) {
+        log_debug(cds)("Skipping %s: lambda proxy class", k->name()->as_C_string());
```

Comment:
Do we need to say why we are skipping it - that it is because of the nest-host?

Reply:
I changed the log statement so that it prints the following with the test case:

`[1.999s][debug  ][cds] Skipping ChildOldInf$InnerChild$$Lambda$1+0x0000000801001000: lambda proxy class because its nest host ChildOldInf was excluded`